File: Easy/twoSum.py
from typing import List

class SolutionO1:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}
        n = len(nums)

        for i in range(n):
            complement = target - nums[i]
            if complement in numMap:
                return [numMap[complement], i]
            numMap[nums[i]] = i

        return []  # No solution found

class SolutionOn2:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        for i in range(n - 1):
            for j in range(i + 1, n):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return [] 
    
twoSumObj = SolutionOn2() 
print("O(n^2) solution {}",twoSumObj.twoSum([2,3,4,5], 5))

twoSumObj = SolutionO1() 
print("O(1) solution {}",twoSumObj.twoSum([2,3,4,5], 5))

File: Easy/twoSum.java
package Easy;

import java.util.*;

class SolutionO1 {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            int complement = target - nums[i];
            if (numMap.containsKey(complement)) {
                return new int[]{numMap.get(complement), i};
            }
            numMap.put(nums[i], i);
        }

        return new int[]{}; // No solution found
    }
}

class SolutionOn2 {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;

        for (int i = 0; i < n-1; i++) {
           for(int j = i+1; j < n; j++){
            if (nums[i] + nums[j] == target){
                return new int[]{i,j};
            }
           }
        }

        return new int[]{}; // No solution found
    }
}

class twoSum{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter number of elements");
        int numElement = sc.nextInt();
        
        int[] nums = new int[numElement];

        System.out.println("Enter elements with space");
        for (int i = 0; i < numElement; i++) {
            nums[i] = sc.nextInt();
        }
        System.out.println("Enter Target");
        int target = sc.nextInt();
        

        SolutionOn2 oN2 = new SolutionOn2();
        SolutionO1 o1 = new SolutionO1();

        System.out.println("O(n2) solution " + Arrays.toString(oN2.twoSum(nums, target)));
        System.out.println("O(1) solution " + Arrays.toString(o1.twoSum(nums, target)));
        sc.close();
    }
}